### PR TITLE
Use reflect.StructTag.Get in fieldtag logic.

### PR DESCRIPTION
**Pull request by PurelyApplied**
It looks like `StructTag` is just a named `string` and we can co-opt `StructTag.Get`.

While `go vet -structtag` doesn't complain about it, it does interact strangely with escaped tags like `"levee:\"source\""`.  But it is much cleaner to use it as the community standard truth.

Opening this PR as a draft until conflicting PRs are merged.

### CODE REVIEW

File: internal/pkg/fieldtags/testdata/src/tests/test.go
Comment (PurelyApplied) [inline internal/pkg/fieldtags/testdata/src/tests/test.go:21]: Exported fields to satisfy `go vet -structtag`.

File: internal/pkg/fieldtags/analyzer.go
Comment (mlevesquedion) [inline internal/pkg/fieldtags/analyzer.go:98]: I think it would be good to have that comment higher up in the method, since if I understand correctly it explains why we want to return if `len(tag) <= 2`.

File: internal/pkg/fieldtags/analyzer.go
Comment (mlevesquedion) [inline internal/pkg/fieldtags/analyzer.go:100]: This is very nice. Glad we're getting rid of the mess above!

I am just now realizing that this does lock us in to using explicit strings, though, at least for the `Key`. I think that is fine, just something we will need to indicate clearly in the documentation.

File: internal/pkg/fieldtags/testdata/src/tests/test.go
Comment (mlevesquedion) [inline internal/pkg/fieldtags/testdata/src/tests/test.go:20]: I think we should still support this case, since `structtag` doesn't complain about it and `StructTag.Get` supports it, e.g. see the example program below. If we use `strconv.Unquote` we can convert the string literal from the `ast` into something `StructTag.Get` can handle, e.g. `"levee:\"source\""` becomes `levee:"source"` with `Unquote`.

Example program showing how StructTag handles the case above (`f.Tag` is a `StructTag`):
```go
package main

import (
	"fmt"
	"reflect"
)

func main() {
	f, _ := reflect.TypeOf(myStruct{}).FieldByName("myField")
	fmt.Println(f.Tag)
}

type myStruct struct {
	myField string "key:\"value\""
}
```
(this prints `key:"value"`)

**Review by mlevesquedion — CHANGES_REQUESTED** [review]
I think using `StructTag` makes sense. Good stuff!

File: internal/pkg/fieldtags/testdata/src/tests/test.go
Comment (PurelyApplied) [inline internal/pkg/fieldtags/testdata/src/tests/test.go:21]: This case changed because we do not support multiple `levee:` values.  We could, but it would need to be accounted for above.

File: internal/pkg/fieldtags/testdata/src/tests/test.go
Comment (PurelyApplied) [inline internal/pkg/fieldtags/testdata/src/tests/test.go:20]: This PR yields no longer supporting these escaped-quote style of tags.  We could jump through some hoops to do so if necessary, but if the point was to keep the lookup simple, I'd rather wait until someone complains that try to preempt the issue.

**Comment by PurelyApplied** [comment]
Sorry for the forced push, but the changes from the initial were sweeping enough that an entire review is rather necessary at this point.

File: internal/pkg/fieldtags/testdata/src/tests/test.go
Comment (PurelyApplied) [reply internal/pkg/fieldtags/testdata/src/tests/test.go]: I didn't know about `strconv.Unquote`!  That keeps things nice and tidy then.

File: internal/pkg/fieldtags/testdata/src/tests/test.go
Comment (mlevesquedion) [reply internal/pkg/fieldtags/testdata/src/tests/test.go:21]: I think I'd prefer >2 values with the target value in the middle or at the end (to test that we're actually looping), but this is fine.

**Review by mlevesquedion — COMMENTED** [review]
I think this looks good. I like how you made the Config take the whole tag, I think that makes sense.

Waiting to see how it looks with `strconv.Unquote` before I approve. I think it'll look cleaner + allow us to handle the case that we would be dropping otherwise, but I may be wrong.

**Comment by PurelyApplied** [comment]
> I'll get to it tomorrow.
What a hopeful person I was two weeks ago.

File: internal/pkg/config/config.go
Comment (mlevesquedion) [inline internal/pkg/config/config.go:59]: Never occurred to me to do something like that (writing to a parameter). Reads well, I like it.

File: internal/pkg/config/fieldtags_test.go
Comment (mlevesquedion) [inline internal/pkg/config/fieldtags_test.go:61]: I think it would be good to have a test case that covers the `for _, v := range strings.Split(val, ",") {` code, now that the logic for multi-value tags is in config.

**Comment by mlevesquedion** [comment]
Haha, you couldn't have known. It's good to be optimistic.

I think this looks very good!

One last thing: I think we need a test for a multi-valued tag in `config/fieldtags_test.go`.

File: internal/pkg/config/fieldtags_test.go
Comment (mlevesquedion) [inline internal/pkg/config/fieldtags_test.go:72]: The description does not seem to match the data in the test case. The data is the same as the next case, though the description is different.

**Comment by mlevesquedion** [comment]
Good extra cases. LGTM, expect for one case where the description does not seem to match the data. The description says "no target" but there is a target and the expected result is `true`.

**Review by mlevesquedion — APPROVED** [review]
Approved, assuming you will fix the test case that has a typo.

File: internal/pkg/config/fieldtags_test.go
Comment (PurelyApplied) [reply internal/pkg/config/fieldtags_test.go:72]: Also, the same test case twice in a row.

Copy-Paste, my old nemesis...

**Pull request merged**
